UnleashClient/connectors/base_connector.py

```diff
@@ -0,0 +1,63 @@
+import json
+from abc import ABC, abstractmethod
+from typing import Callable, Optional
+
+from yggdrasil_engine.engine import UnleashEngine
+
+from UnleashClient.cache import BaseCache
+from UnleashClient.constants import FEATURES_URL
+from UnleashClient.utils import LOGGER
+
+
+class BaseConnector(ABC):
+    def __init__(
+        self,
+        engine: UnleashEngine,
+        cache: BaseCache,
+        ready_callback: Optional[Callable] = None,
+    ):
+        """
+        :param engine: Feature evaluation engine instance (UnleashEngine).
+        :param cache: Should be the cache class variable from UnleashClient
+        :param ready_callback: Optional function to call when features are successfully loaded.
+        """
+        self.engine = engine
+        self.cache = cache
+        self.ready_callback = ready_callback
+
+    @abstractmethod
+    def start(self):
+        pass
+
+    @abstractmethod
+    def stop(self):
+        pass
+
+    def load_features(self):
+        feature_provisioning = self.cache.get(FEATURES_URL)
+        if not feature_provisioning:
+            LOGGER.warning(
+                "Unleash client does not have cached features. "
+                "Please make sure client can communicate with Unleash server!"
+            )
+            return
+
+        try:
+            payload = (
+                json.dumps(feature_provisioning)
+                if isinstance(feature_provisioning, dict)
+                else feature_provisioning
+            )
```

Comment:
What's the need for coercing the payload here? We should always have a string, shouldn't we?